community/modules/compute/notebook/main.tf

```diff
@@ -48,21 +48,22 @@ resource "random_id" "resource_name_suffix" {
   byte_length = 4
 }
 
+resource "google_storage_bucket_object" "mount_script" {
+  name    = local.post_script_filename
+  content = local.content5
+  bucket  = local.bucket
+}
+
 resource "google_notebooks_instance" "instance" {
   name                = local.name
   location            = var.zone
   machine_type        = var.machine_type
   project             = var.project_id
   post_startup_script = "${var.gcs_bucket_path}/${local.post_script_filename}"
```

Comment:
I think another, arguably better, way to accomplish what you are doing with `depends_on` would be change this line to be:

```
# old
post_startup_script = "${var.gcs_bucket_path}/${local.post_script_filename}"
# new
post_startup_script = "${var.gcs_bucket_path}/${google_storage_bucket_object.mount_script.name}"
```

`depends_on` tends to be the nuclear option and can have some undesirable behavior such as destroying resources when really they could just be updated in place.

LMK if you are opposed to the suggestion. I am not set in stone on this.

Reply:
I'm not against it and happy to do it. I just don't know of any way to test whether it actually solves to proposed problem. I know `depends_on` is designed to ensure the resource is there. 

If you are comfortable with the proposed change, I have nothing to base an objection upon!